vidplayer/player.go

```diff
@@ -63,28 +65,34 @@ func handleHLS(w http.ResponseWriter, r *http.Request, getHLSBuffer func(reqPath
 	}
 
 	if strings.HasSuffix(r.URL.Path, ".m3u8") {
+		glog.Infof("Before waitAndPopPlaylist: %v", time.Now())
 		pl, err := buffer.WaitAndPopPlaylist(ctx)
+		// pl, err := buffer.LatestPlaylist()
+		glog.Infof("After waitAndPopPlaylist: %v", time.Now())
 		if err != nil {
 			glog.Errorf("Error getting HLS playlist %v: %v", r.URL.Path, err)
 			return
 		}
 
-		//Remove all but the last 2 segments
+		//Remove all but the last 5 segments
 		c := 0
 		for _, seg := range pl.Segments {
 			if seg != nil {
 				// segs = append(segs, seg)
 				c = c + 1
 			}
 		}
-		for c > 2 {
+		for c > 5 {
```

Comment:
Segment length and the player buffer length are probably good candidates for constant variables instead of hardcoded here since we probably will want to reuse them throughout LPMS.